vllm/lora/punica.py

```diff
@@ -5,17 +5,30 @@
 import torch
 
 from vllm import _custom_ops as ops
+from vllm.logger import init_logger
+from vllm.lora import native_kernels
+from vllm.utils import is_cpu
+
+logger = init_logger(__name__)
+
+if is_cpu():
+    logger.warning(
+        "punica LoRA kernels on CPU will be fallback to Torch native kernels, "
+        "which will cause non-optimized performance.")
 
 
 def _check_punica_support():
+    if is_cpu():
+        return native_kernels
+
     if ops.is_custom_op_supported("_punica_C::dispatch_bgmv"):
-        return
+        return ops
 
     if torch.cuda.get_device_capability() < (8, 0):
         raise ImportError(
             "punica LoRA kernels require compute capability >= 8.0")
```

Comment:
Seems you also want to enable the LoRA for CUDA devices < 8.0. If it is a goal of this PR, should we also handle this branch similarly to the CPU and issue a warning?

Reply:
Oh, I forgot to handle this part in the rebase. Thanks for reminding!